be/src/vec/olap/block_reader.h

```diff
@@ -110,6 +114,8 @@ class BlockReader final : public TabletReader {
     ColumnPtr _delete_filter_column;
 
     bool _is_rowsets_overlapping = true;
+
+    bool _should_check_same_row = false;
```

Comment:
don't we consider VerticalBlockReader?

Reply:
In vertical block reader, we get next block directly instead of get row by row, so it's hard to achieve this row same check

Reply:
The key compare logic is same in key group of vertical compaction?